saleor/plugins/openid_connect/utils.py

```diff
@@ -245,12 +270,43 @@ def get_user_from_oauth_access_token(
             "Failed to fetch OIDC user info", extra={"user_info_url": user_info_url}
         )
         return None
-    user = get_or_create_user_from_payload(user_info, oauth_url=user_info_url)
-    if not use_scope_permissions:
+    user = get_or_create_user_from_payload(
+        user_info,
+        oauth_url=user_info_url,
+    )
+
+    email_domain = get_domain_from_email(user.email)
+    is_staff_email = email_domain in staff_user_domains
+    if not use_scope_permissions and not is_staff_email:
         user.is_staff = False
+    elif is_staff_email:
+        assign_staff_to_default_group_and_update_permissions(
+            user, staff_default_group_name
+        )
+
     return user
 
 
+def assign_staff_to_default_group_and_update_permissions(
+    user: "User", default_group_name: str
+):
+    """Assign staff user to the default permission group. and update user permissions.
+
+    If the group doesn't exist, the new group without any assigned permissions or
+    channels.
+    """
+    default_group_name = (
+        default_group_name.strip() if default_group_name else default_group_name
```

Comment:
Just want to check for sure, that this `if` here is purposely. Cause right now we will accept names like "   ", but not "".